_test/test_sqw_pageOpMethods/test_split.m

```diff
@@ -0,0 +1,157 @@
+            end
+        end
+
+
+        function test_split_filebacked_eq_split_membased(obj)
+
+            w_spl_mem = split(obj.source_sqw4D);
+
+            n_pix = obj.source_sqw4D.npixels;
+
+            clConf = set_temporary_config_options(hor_config,'mem_chunk_size',n_pix/3);
+            source = sqw(obj.sqw_source,'file_backed',true);
+            assertTrue(source.is_filebacked);
+
+            w_spl_fb = split(source,'-filebacked');
+
+            assertEqual(numel(w_spl_mem),23);
+            clear clConf
+
+
+            for i=1:numel(w_spl_mem)
+                assertEqualToTol(w_spl_mem(i),w_spl_fb(i),'ignore_str',true, ...
+                    'tol',[8*eps('single'),8*eps('single')]);
+            end
+        end
+
+        function test_split_pix_filebacked(obj)
+            n_pix = obj.source_sqw4D.npixels;
+            clConf = set_temporary_config_options(hor_config,'mem_chunk_size',n_pix/3);
+            source = sqw(obj.sqw_source,'file_backed',true);
+            assertTrue(source.is_filebacked);
+
+            w_spl = split(source,'-filebacked');
+
+            assertEqual(numel(w_spl),23);
+
+            n_split_pix = 0;
+            for i=1:numel(w_spl)
+                assertTrue(w_spl(i).is_filebacked);
+                keys = w_spl(i).runid_map.keys;
+                assertEqual(numel(keys),1);
+                id = unique(w_spl(i).pix.run_idx);
+                assertEqual(keys{1},id);
+                assertEqual(w_spl(i).experiment_info.expdata.run_id,id);
+                n_split_pix  = n_split_pix +w_spl(i).npixels;
+            end
+            assertEqual(n_pix,n_split_pix);
+
+            % check that filebacked objects are temporary objects in this
+            % case
+            files = cell(1,numel(w_spl));
+            for i=1:numel(files)
+                files{i} = w_spl(i).pix.full_filename;
+                assertTrue(is_file(files{i}));
+            end
+            clear('w_spl');
+            for i=1:numel(files)
+                assertFalse(is_file(files{i}));
+            end
+        end
+
+        function test_split_all_in_memory(obj)
+            n_pix = obj.source_sqw4D.npixels;
+
+            w_spl = split(obj.source_sqw4D);
+
+            assertEqual(numel(w_spl),23);
+
+            n_split_pix = 0;
+            for i=1:numel(w_spl)
+                keys = w_spl(i).runid_map.keys;
+                assertEqual(numel(keys),1);
+                id = unique(w_spl(i).pix.run_idx);
+                assertEqual(keys{1},id);
+                assertEqual(w_spl(i).experiment_info.expdata.run_id,id);
+                n_split_pix  = n_split_pix +w_spl(i).npixels;
+            end
+            assertEqual(n_pix,n_split_pix);
+        end
+
+        function test_prepare_split_sqw_in_mem(obj)
+            page_op = PageOp_split_sqw();
+
+            page_op = page_op.prepare_split_sqw(obj.source_sqw4D,false);
+            n_runds = obj.source_sqw4D.main_header.nfiles;
```

Comment:
`n_runs`?

Reply:
well -- file is run. Everywhere in Horace

Reply:
No, my point is you have made a typo and written `n_runds`